hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/BlockCacheUtil.java

```diff
@@ -244,6 +248,45 @@ public static int getMaxCachedBlocksByFile(Configuration conf) {
     return conf == null ? DEFAULT_MAX : conf.getInt("hbase.ui.blockcache.by.file.max", DEFAULT_MAX);
   }
 
+  /**
+   * Similarly to HFileBlock.Writer.getBlockForCaching(), creates a HFileBlock instance without
```

Comment:
Could we share some common code between these two methods? Otherwise how can we align these two methods?

Reply:
Yes, let me work on this.

Reply:
So the most I could reuse here is the new HFIleContext part. The block builder portion is hard to reuse because in HFileBlock.Writer the parameters are calculated on the go, so any sort of reuse would require an awful lot of parameters.